In [1]:
import os
for f in os.listdir('../fun/'): exec(open('../fun/'+f).read())
del f

# Load data.
load( '../out/d3-fight-level-transform.pkl' )
print( X.shape )

(4091, 166)


In [2]:
# train test split.
from sklearn.model_selection import train_test_split
X_train , X_test, y_train, y_test = train_test_split(
    X, y, 
    random_state = 718,
    test_size = 0.25
)

In [17]:
# Fit decision tree.
from sklearn.tree import DecisionTreeClassifier
m = DecisionTreeClassifier( 
    max_depth = 20, 
    min_samples_split = .03,
    random_state = 716 
)
m.fit( X_train, y_train )

# accuracy against train data (in-model).
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_train, m.predict(X_train) ) )
# prior precision/fscore (before adding odds): .76 / .81

              precision    recall  f1-score   support

           0       0.73      0.66      0.69      1033
           1       0.84      0.88      0.86      2035

    accuracy                           0.80      3068
   macro avg       0.78      0.77      0.77      3068
weighted avg       0.80      0.80      0.80      3068



In [19]:
# accuracy against test data.
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_test, m.predict(X_test) ) )
# prior precision/fscore (before adding odds): .74 / .76

              precision    recall  f1-score   support

           0       0.47      0.41      0.44       355
           1       0.71      0.75      0.73       668

    accuracy                           0.64      1023
   macro avg       0.59      0.58      0.59      1023
weighted avg       0.63      0.64      0.63      1023



In [6]:
from sklearn.model_selection import cross_validate
kfold = cross_validate( 
    m, X, y, 
    cv = 10, 
    scoring = [ 'precision', 'recall' ], 
    n_jobs = -1 
)
print( 
  'Mean precision: %s \nMean recall %s' % 
  ( kfold['test_precision'].mean(), kfold['test_recall'].mean() ) 
)
# prior precision/racall (before adding odds): .69 / .73

Mean precision: 0.6898793655075857 
Mean recall 0.7371586715867158


f1-score in-model is higher than on test indicating **over-fitting**. 

Test-set precision is still not great at 69% (predicting all-wins give 60%).